rest/config_legacy.go

```diff
@@ -318,6 +319,11 @@ func LoadServerConfig(path string) (config *LegacyServerConfig, err error) {
 // readServerConfig returns a validated LegacyServerConfig from an io.Reader
 func readServerConfig(r io.Reader) (config *LegacyServerConfig, err error) {
```

Comment:
as discussed, rename to reflect the struct it returns

```suggestion
func readLegacyServerConfig(r io.Reader) (config *LegacyServerConfig, err error) {
```